core/arch/aarch64/codec.c

```diff
@@ -1128,12 +1128,16 @@ encode_opnd_imm5(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_out)
 static inline bool
 decode_opnd_imm6(uint enc, int opcode, byte *pc, OUT opnd_t *opnd)
 {
+    if (!TEST(1U << 31, enc) && TEST(1U << 15, enc))
+        return false;
```

Comment:
Not sure I see where this is coming from.  This is a bug fix?  Should be mentioned in the commit message?  Same below.